### PR TITLE
Ignore irrelevant file changes on azure

### PR DESCRIPTION
**Pull request by nilsnolde**
azure is again pretty curious about design choices, e.g. wildcarding file patterns doesn't work..

also added the other CI configs to circle's `ciignore` list.

### CODE REVIEW

**Comment by nilsnolde** [comment]
btw @mandeepsandhu couldn't we also disable the docker builds on circle for those irrelevant changes?

**Comment by nilsnolde** [comment]
hm.. maybe it has to run once successfully all the way with the changes to ci config.. then update md, then trigger again

**Comment by nilsnolde** [comment]
azure still builds happily.. if SO doesn't get me any help, I consider it shelved for now: https://stackoverflow.com/questions/67138364/azure-devops-pipeline-exclude-paths-for-prs

**Comment by nilsnolde** [comment]
I really like that azure skips draft PRs. I had a look for circle ci and there is a public extension which looks half legit: https://circleci.com/developer/orbs/orb/artsy/skip-wip-ci. In the config one of its commands would be a step, e.g.
```
version: 2.1
orbs:
  skip-wip-ci: artsy/skip-wip-ci@0.3.0
....
    steps:
      - checkout
      - skip-wip-ci/check-skippable-pr
```

worth a try or overzealous?

**Comment by nilsnolde** [comment]
I just tried to do the same as mentioned here: https://github.com/valhalla/valhalla/pull/2999#issuecomment-822519578

luckily azure behaves properly and builds again even though it's only been a md change in the last commit.

**Review by mloskot — APPROVED** [review]

**Comment by nilsnolde** [comment]
Can I merge @kevinkreiser ?

**Comment by nilsnolde** [comment]
failing bcs of #3037 I guess

File: CHANGELOG.md [hunk 1]
```diff
@@ -107,7 +107,8 @@
    * CHANGED: nit(rename): Renames the encoded live speed properties [#2998](https://github.com/valhalla/valhalla/pull/2998)
    * ADDED: ci: Vendors the codecov script [#3002](https://github.com/valhalla/valhalla/pull/3002)
    * CHANGED: Allow None build type [#3005](https://github.com/valhalla/valhalla/pull/3005)
-   * CHANGED: ci: Build Python bindings for Mac OS [#3011](https://github.com/valhalla/valhalla/pull/3011)
+   * CHANGED: ci: Build Python bindings for Mac OS [#3013](https://github.com/valhalla/valhalla/pull/3013)
+   * CHANGED: ci: Skip Win CI runs for irrelevant files [#3014](https://github.com/valhalla/valhalla/pull/3014)
```
Comment (kevinkreiser) [inline CHANGELOG.md:111]: @nilsnolde this is in the wrong section now since we cut a release a week ago. can you move this, than we can merge

Comment (nilsnolde) [reply CHANGELOG.md]: jep done:)

**Comment by nilsnolde** [comment]
shouldn't azure skip the CI for a simple changelog commit.. :thinking:

**Comment by mandeepsandhu** [comment]
> shouldn't azure skip the CI for a simple changelog commit..

@nilsnolde CI will skip only if the previous run had ALL tests passing. Looking at status of https://github.com/valhalla/valhalla/pull/3014/commits/ca5c56f408b952e1fe72113141fc1299ce5ef6e7 the osx build failed for it, so we didn't skip the CI.

If osx is optional, we can remove it from the [list of builds](https://github.com/valhalla/valhalla/blob/master/scripts/needs_ci_run#L15) that the skip CI script expects to pass.

**Comment by kevinkreiser** [comment]
@mandeepsandhu we dont want to be optional its just still screwed up at the moment. do you know anyone with a macbook who would be willing to scrifice their time to getting it working again? if not i'll have to ssh into the machine with circleci and play around with it. we should make an issue for getting that sorted out. we dont want to run too long without actively making sure osx builds for sure

**Comment by mandeepsandhu** [comment]
> @mandeepsandhu we dont want it to be optional its just still screwed up at the moment. do you know anyone with a macbook who would be willing to scrifice their time to getting it working again? if not i'll have to ssh into the machine with circleci and play around with it. we should make an issue for getting that sorted out. we dont want to run too long without actively making sure osx builds for sure

I opened https://github.com/valhalla/valhalla/issues/3036 to get off of older homebrew version which was trying to use bintray. However, the [new homebrew version is having its own issue](https://github.com/valhalla/valhalla/pull/3037) with the prime_server cask. I'll ask around if someone with a mac can help debug this further.

**Comment by nilsnolde** [comment]
> CI will skip only if the previous run had ALL tests passing.

Not sure if Azure cares about failing circle CI? they should be acting pretty independently.. and as far as azure was concerned https://github.com/valhalla/valhalla/pull/3014/commits/ca5c56f408b952e1fe72113141fc1299ce5ef6e7 was (successfully) run and https://github.com/valhalla/valhalla/pull/3014/commits/ff4dab99fc998a7c5c9b308806362f572f702120 simply changed a little markdown, yet it ran again.. mabe I'm underestimating the CI magic going on

**Comment by nilsnolde** [comment]
Sorry, I missed merging this after the last approval.. Could you approve again pls @kevinkreiser , feel free to merge yourself of course.

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**